### PR TITLE
feat: Multi-step indicator DOCS-470

### PR DESCRIPTION
**Pull request by nicklem**
Adds a multi-step indicator and nav menu to help users navigate between the pages of the onboarding flow.

### :eyes: Live preview

https://feat-multi-step-indicator-docs-470--docs-codacy.netlify.app/getting-started/codacy-quickstart/#adding-your-first-repository

### :construction: To do
- Before merging this:
  - [x] Merge https://github.com/codacy/codacy-mkdocs-material/pull/79
  - [x] Update submodule `codacy-mkdocs-material`

### CODE REVIEW

**Comment by nicklem** [comment]
We may also want to mention in a sentence or two that this is a multi-page onboarding process. Thoughts?

**Comment by mrfyda** [comment]
hey team 👋 

this looks like a nice improvement to the page!

I also wanted to share with you that we have an ongoing pulse initiative to bring the same page to the product itself: https://codacy.atlassian.net/browse/PUL-1708
if you think this something you'd like to leverage for Quality, I'm happy to help

**Comment by prcr** [comment]
> hey team wave
> 
> this looks like a nice improvement to the page!
> 
> I also wanted to share with you that we have an ongoing pulse initiative to bring the same page to the product itself: https://codacy.atlassian.net/browse/PUL-1708 if you think this something you'd like to leverage for Quality, I'm happy to help

Thanks for checking this out @mrfyda! :heart: 

We actually [created a Jira issue](https://codacy.atlassian.net/browse/DOCS-506) for implementing the same thing on Codacy Quality after I saw your idea on Slack. :wink: We'll ask for help when we're ready to start working on that. :bow:

**Comment by prcr** [comment]
@nicklem I pushed [this "WIP" commit](https://github.com/codacy/docs/pull/1606/commits/9114e7154a2b89a08d8e297d3783ddb1741a6e45) updating the codacy-mkdocs-material submodule to the head commit of the branch `feat/multi-step-indicator-styles-DOCS-470` so that we could see the full proposal on the pull request preview. :star_struck:

(After merging https://github.com/codacy/codacy-mkdocs-material/pull/79 you'll need to update the submodule again so it points to the last commit in `master`.)

File: docs/assets/includes/nav-multistep.md
Comment (prcr) [inline docs/assets/includes/nav-multistep.md:1]: I suggest giving a more specific name to this "include file", to take into account that we may have more guided paths like this one in the future. For example, `nav-multistep-quickstart.md` or `nav-repository-onboarding.md`.

File: docs/assets/includes/nav-multistep.md
Comment (prcr) [inline docs/assets/includes/nav-multistep.md:14]: On the other hand, the meta fields could have more generic names (so we can more easily copy and reuse this pattern for other uses):

```suggestion
        <li class="{% if page.meta.nav_step == 0 %}nav-multistep__step--active{% endif %}">
            <a href="/getting-started/codacy-quickstart#adding-your-first-repository">
                Adding your first repository
            </a>
        </li>
        <li class="{% if page.meta.nav_step == 1 %}nav-multistep__step--active{% endif %}">
            <a href="/getting-started/configuring-your-repository#configuring-your-repository">
                Configuring your repository
            </a>
        </li>
        <li class="{% if page.meta.nav_step == 2 %}nav-multistep__step--active{% endif %}">
            <a href="/getting-started/integrating-codacy-with-your-git-workflow#integrating-codacy-with-your-git-workflow">
```

File: docs/assets/includes/nav-multistep.md
Comment (prcr) [inline docs/assets/includes/nav-multistep.md:1]: I think we should experiment with including a small introductory sentence before the navigation, for example:

```suggestion
This page is step {{ page.meta.onboarding_step }} of the following guided path:

<nav class="nav-multistep">
```

Or a simpler version that doesn't mention the step:

```suggestion
This page is part of the following guided path:

<nav class="nav-multistep">
```

Comment (nicklem) [reply docs/assets/includes/nav-multistep.md]: Agreed. I updated the name to `nav-multistep-quickstart.md`.

File: docs/assets/includes/nav-multistep.md
Comment (nicklem) [reply docs/assets/includes/nav-multistep.md]: Done 👍

File: docs/assets/includes/nav-multistep.md
Comment (nicklem) [reply docs/assets/includes/nav-multistep.md]: Agreed. Done 👍

**Comment by nicklem** [comment]
@prcr Ready for approval

**Comment by nicklem** [comment]
@prcr @claudiacarpinteiro 

Please feel free to approve https://github.com/codacy/docs/pull/1610 and then merge this one if all looks good! 😄 

I'll be signing off for today.

**Comment by prcr** [comment]
@nicklem I pushed this one last simplification https://github.com/codacy/docs/pull/1606/commits/73a91fabc9c3468ddc4b77a6038baae58fb42f5d, please check it out. I'm now merging the changes. :star_struck: :rocket:

**Review by prcr — APPROVED** [review]
Excellent solution to have this new pattern in the docs and ready for future uses! :raised_hands:

**Comment by prcr** [comment]
I did [this force-push](https://github.com/codacy/docs/compare/73a91fabc9c3468ddc4b77a6038baae58fb42f5d..fc6ee6395649b117138d1f8e16310e54f56cd362) to try to unblock the Codacy status check.

**Pull request merged**
